File: 136-single-number/single-number.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int singleNumber(int[] nums) {

        HashMap<Integer, Integer> freq = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {

            if (!freq.containsKey(nums[i])) {
                freq.put(nums[i], 1);
            } else {
                freq.put(nums[i], freq.get(nums[i]) + 1);
            }
        }

        for (Map.Entry<Integer, Integer> entry : freq.entrySet()) {

            if (entry.getValue() == 1) {
                return entry.getKey();
            }
        }

        return -1;
    }
}

File: 136-single-number/single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        freq = {}
        for i in range(len(nums)):
            if nums[i] not in freq:
                freq[nums[i]] = 1
            else:
                freq[nums[i]] += 1

        for key, value in freq.items():
            if value == 1:
                return key